js/components/addEditBookmark.js

```diff
@@ -89,11 +89,12 @@ class AddEditBookmark extends ImmutableComponent {
     windowActions.setBookmarkDetail(currentDetail, this.props.originalDetail, this.props.destinationDetail)
   }
   showToolbarOnFirstBookmark () {
-    const showBookmarksToolbar = getSetting(settings.SHOW_BOOKMARKS_TOOLBAR)
-    const isFirstBookmark = this.props.sites.find(
+    const hasOneBookmark = this.props.sites.find(
       (site) => siteUtil.isBookmark(site) || siteUtil.isFolder(site)
     )
-    appActions.changeSetting(settings.SHOW_BOOKMARKS_TOOLBAR, !isFirstBookmark || showBookmarksToolbar)
+    if (!hasOneBookmark && !getSetting(settings.SHOW_BOOKMARKS_TOOLBAR)) {
+      appActions.changeSetting(settings.SHOW_BOOKMARKS_TOOLBAR, true)
+    }
   }
```

Comment:
These changes were made to prevent appActions.changeSetting() from being called every time. Now it will only call if this is your first bookmark and toolbar is not already visible